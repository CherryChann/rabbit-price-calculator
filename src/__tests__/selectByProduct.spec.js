import React from 'react';
import { shallow, mount } from 'enzyme';
import SelectByProduct from '../components/selectByProduct';
import { Row, Col, Form} from 'react-bootstrap';

const onClick = (value) => { 
    console.log(value, 'mock function');
}
const props = {
    products: [
        {
            id: 1,
            name: 'product 1'
        },
        {
            id: 2,
            name: 'product 2'
        }
    ],
    onClick: onClick
}

describe('Shallow rendered select by product component ', () => {
    const container = shallow(
        <SelectByProduct
            products={props.products}
            selectedProduct={props.products[1]}
            onClick={props.onClick}
        />);
    it('should render select box with products', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should have correct props, called props function and fire selected product', () => {
        expect(container.find('.product-selection').prop('as')).toEqual('select');
        expect(container.find('option').length).toEqual(3); // including placeholder option and products
        container.find('.product-selection').simulate('change', {
            target: {
                value: 2
            }
        })
        expect(container.find('.product-selection').prop('value')).toEqual(2);
    })

    it('should select placeholder if selected product is not fired', () => {
        const container = shallow(
        <SelectByProduct
            products={props.products}
            onClick={props.onClick}
        />);
        expect(container.find('.product-selection').prop('value')).toEqual('placeholder');
    })
});