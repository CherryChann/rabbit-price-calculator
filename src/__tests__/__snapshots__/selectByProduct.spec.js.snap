// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered select by product component  should render select box 1`] = `"<div class=\\"margin-top row\\"><div class=\\"col-lg-2 col-12\\"><span>Products: </span></div><div class=\\"col-lg-3 col-12\\"><div class=\\"form-group\\"><select class=\\"product-selection form-control\\"><option value=\\"placeholder\\"> Select Products...</option><option value=\\"1\\">product 1</option><option selected=\\"\\" value=\\"2\\">product 2</option></select></div></div></div>"`;
